ctk/common/src/main/kotlin/org/codice/compliance/verification/core/ResponseProtocolVerifier.kt

```diff
@@ -15,160 +15,105 @@ package org.codice.compliance.verification.core
+        CommonDataTypeVerifier.verifyStringValues(version)
         CommonDataTypeVerifier.verifyDateTimeValues(
-                response.attributes.getNamedItem(ISSUE_INSTANT), SAMLCore_3_2_2_d)
-
-        val destination = response.attributes?.getNamedItem("Destination")?.textContent
-        if (destination != acsUrl)
-            throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_3_2_2_e,
-                    property = "Destination",
-                    actual = destination,
-                    expected = acsUrl ?: "No ACS URL Found",
-                    node = response)
+                response.attributes.getNamedItem("IssueInstant"), SAMLCore_3_2_2_d)
+
+        response.attributes?.getNamedItem(DESTINATION)?.let {
+            if (it.textContent != acsUrl)
+                throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_3_2_2_e,
+                        property = DESTINATION,
+                        actual = it.textContent,
+                        expected = acsUrl ?: "No ACS URL Found",
+                        node = response)
 
-        if (response.children(STATUS).isEmpty())
-            throw SAMLComplianceException.createWithXmlPropertyReqMessage(SAMLCore_3_2_2,
-                    property = STATUS,
-                    parent = RESPONSE,
-                    node = response)
+            CommonDataTypeVerifier.verifyUriValues(it)
+        }
 
         CoreVerifier.verifySamlExtensions(response.children(),
                 expectedSamlNames = listOf("Issuer", "Signature", "Status", "Assertion",
                         "EncryptedAssertion"))
-    }
 
-    /**
-     * Verify the Statuses and Status Codes
-     * 3.2.2.1 Element <Status>
-     * 3.2.2.2 Element <StatusCode>
-     */
-    private fun verifyStatusesType() {
-        // Status
-        response.children(STATUS).forEach {
-            val statusCodes = it.children(STATUS_CODE)
-            if (statusCodes.isEmpty())
-                throw SAMLComplianceException.createWithXmlPropertyReqMessage("SAMLCore.3.2.2.1",
-                        property = STATUS_CODE,
-                        parent = STATUS,
-                        node = response)
+        response.attributes?.getNamedItem("Content")?.let {
+            CommonDataTypeVerifier.verifyUriValues(it)
+        }
+    }
 
-            // StatusCode
-            if (statusCodes.any { it.attributes.getNamedItem(VALUE) == null })
-                throw SAMLComplianceException.createWithXmlPropertyReqMessage("SAMLCore.3.2.2.2",
-                        property = VALUE,
-                        parent = STATUS_CODE,
-                        node = response)
+    /** 3.2.2.2 Element <StatusCode> */
+    private fun verifyStatusType() {
+        if (response.recursiveChildren("Status")
+                        .any {
+                            !TOP_LEVEL_STATUS_CODES
+                                    .contains(it.children(STATUS_CODE).first()
```

Comment:
❓ Do we need to protect against the case where there are no status code nodes?

Reply:
`StatusCode` is a required element of `Status`